docs/public-networks/how-to/bonsai-limit-trie-logs.md

```diff
@@ -0,0 +1,187 @@
+---
+title: Limit Trie Logs for Bonsai
+sidebar_position: 12
+description: Enable this early access feature to reduce the size of your database
+tags:
+  - public networks
+---
+
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+
+# Limit Trie Logs for Bonsai
+## Step by Step Guide
+
+:::caution
+
+We strongly recommend reading the rest of this documentation before running these commands because your node’s configuration may require changes to these commands.
+
+:::
+
+*Targeting 24.1.2 release, but currently you must be on the Besu `main` branch for this to work*
+
+1. Update Besu config to add --Xbonsai-limit-trie-logs-enabled but don’t restart yet
+1. Stop Besu
+1. (optional) Run: 
+`sudo /usr/local/bin/besu/bin/besu --data-storage-format=BONSAI --data-path=/var/lib/besu --sync-mode=X_SNAP storage x-trie-log prune`
+1. Start Besu (remembering to run sudo systemctl daemon-reload if you use a systemd service file as per CoinCashew and Somer)
+1. Look out for `Limit trie logs enabled: retention: 512; prune window: 30000` in your Besu config printout during startup
+1. Enjoy more GBs
+
+## What?
+We have a new experimental feature available: `--Xbonsai-limit-trie-logs-enabled` which aims to keep Besu’s database as small as it can be. After a brief grace period, we intend to make this enabled by default for stakers. This is only relevant if you're using `data-storage-format=BONSAI`.
+
+From our testing, we estimate this will **save users > 3 GB per week** in database growth. Early testing indicates Besu’s **overall database growth with this enabled is ~7 GB per week** (thanks Yorick!) which is on par with geth.
+
+## Why?
+Some users noticed that resyncing Besu can free up disk space, especially for longer running nodes. This is despite `--data-storage-format=BONSAI` having “implicit pruning”. More on this reddit thread: https://www.reddit.com/r/ethstaker/comments/12xnxxi/clearing_up_besubonsai_confusion_on_state_growth/
+
+In reality, whilst the Merkle Patricia Trie is implicitly pruned, the BONSAI feature did introduce an extra data structure: the Trie Log (more detail about BONSAI and trie logs here: https://consensys.io/blog/bonsai-tries-guide)
```

Comment:
I am not sure this is needed in this guide

Reply:
Agree, this is better for the initial announcement